src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -813,7 +842,19 @@ private long getOffset() {
 
       // This api should throw exception if backoff failed.
       // fileNamesCopy after this call is emptied (clears the input list)
-      ingestionService.ingestFiles(fileNamesCopy);
+      if (ingestionDeliveryGuarantee
+          == SnowflakeSinkConnectorConfig.IngestionDeliveryGuarantee.EXACTLY_ONCE) {
+        ingestionService.ingestFilesWithClientInfo(fileNamesCopy, this.clientSequencer.get());
+        String offsetToken = ingestionService.getClientStatus().getOffsetToken();
+        // Update server side offset
+        if (offsetToken == null) {
+          this.offsetPersistedInSnowflake.set(-1);
+        } else {
+          this.offsetPersistedInSnowflake.set(Long.parseLong(offsetToken));
```

Comment:
Just thinking out loud here. Why do we want to set this value after returning from another rest call for getClientStatus? 

Cant we rely on `insertFilesWithClientInfo`'s successful call? Basically if it is a success, the `this.offsetPersistedInSnowflake.set(lastOffsetFromFrom_fileNamesCopy)`

I am okay with one more though for completeness.

Reply:
if we only need the offsetPersistedInSnowflake during rebalancing, this call is kind of redundant already. i am also ok with directly setting the offset to the lastOffsetFromFilename.

Reply:
We can eventually get away with it if we have enough data about its completeness. Can you put a log here?

Reply:
Also, lets verify we have a logger in `insertFilesWithClientInfo's` on what we are passing in as offsetToken.. that way we can decide if this is a redundant call